flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1952,9 +1952,11 @@ class ScalarExprLowering {
     mlir::Value retValBase = fir::getBase(retVal);
     if (fir::conformsWithPassByRef(retValBase.getType()))
```

Comment:
This condition is checked a second time in placeScalarValueInMemory.

Reply:
Fair point, I updated the code to just call `placeScalarValueInMemory` directly.